cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java

```diff
@@ -982,7 +985,9 @@ public final CradleResultSet<StoredMessageBatch> getMessageBatches(MessageFilter
     public final CradleResultSet<StoredGroupedMessageBatch> getGroupedMessageBatches(GroupedMessageFilter filter)
             throws CradleStorageException, IOException {
         logger.debug("Filtering grouped message batches by {}", filter);
-        checkAbstractFilter(filter);
+        if (!checkFilter(filter)) {
+            return new EmptyResultSet<>();
```

Comment:
I think there is no need to create a new instance of `EmptyResultSet` each time it is needed. We can use the same approach here as Java uses for its empty collections